pkg/controller/kubedirectorcluster/members.go

```diff
@@ -239,6 +240,48 @@ func syncMemberNotifies(
 	wgReady.Wait()
 }
 
+// EZML-960
+// extractStartScriptLog extracts the first ~300 symbols
+// from a passed file.
+// This method is used after startscript executions for
+// putting its result to corresponding MemberStateDetail fields
+func extractStartScriptLog(
+	readFileFn func(string, io.Writer) (bool, error),
+	filePath string,
+) *string {
+
+	maxLen := 300
+	var strB strings.Builder
+	fileExists, fileError := readFileFn(filePath, &strB)
+	if fileError != nil {
+		return nil
+	}
+	var msg string
+	if fileExists {
+		msg = shared.TruncateText(strB.String(), maxLen)
```

Comment:
Of course, attempts to do punctuation-aware truncation will get slightly weirder when truncating to the tail rather than the head. I'm sure you could make it work, or it's also fine to be less fancy and just do a simple/hard truncation.

As mentioned above we do (I think) want to have an absolute limit imposed in the schema which we cannot exceed, regardless of what punctuation is in the message.

Reply:
What about to print last few lines instead of characters limit?